develop/toolkit/api-libraries/polkadot-js-sdk.md

```diff
@@ -1,4 +1,109 @@
 ---
 title: Polkadot.js SDK
-description: TODO
----
\ No newline at end of file
+description: Interact with Polkadot SDK-based chains easily using the Polkadot.js SDK. Query chain data, submit transactions, and more via JavaScript.
+---
+
+# Polkadot.js SDK
+
+## Introduction
+
+The Polkadot.js SDK uses JavaScript/Typescript to interact with Polkadot SDK-based chains. It allows you to query nodes, read chain state, and submit transactions through a dynamic, auto-generated API interface.
+
+### Dynamic API Generation
+
+Unlike traditional static APIs, the Polkadot.js API generates its interfaces automatically when connecting to a node. Here's what happens when you connect:
+
+1. The API connects to your node
+2. It retrieves the chain's metadata
+3. Based on this metadata, it creates specific endpoints in the format - `api.<type>.<module>.<section>`
+
+### Available API Categories
+
+You can access three main categories of chain interactions:
+
+1. **Constants** (`api.consts`)
+    - Access runtime constants directly
```

Comment:
What is a "runtime constant"?

Reply:
Agree we should add a short explainer on what each component is

Reply:
Yes and No: you should build on top of existing material: 

- https://paritytech.github.io/polkadot-sdk/master/frame_support/pallet_macros/attr.constant.html

Reply:
I do agree to use existing references, but I don't think the content in that URL is relevant for the developer who wants to use the javascript interfaces exposed by the API. If further explanation or details are needed, I think it is better to reference the following:
- Runtime constants: https://polkadot.js.org/docs/api/start/api.consts
- State queries: https://polkadot.js.org/docs/api/start/api.query
- Transactions: https://polkadot.js.org/docs/api/start/api.tx

Reply:
gladly, those seem like better references.

I do hope that we had a better intermediary page that would explain metadata, and then what are constants in it, but I don't think one exists. 

In the rust docs, we also have https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/metadata/index.html, but it is still a bit too high level.

Reply:
We have a specific page for metadata, but it has yet to be mapped to the new informational architecture. You can check the content here: https://github.com/polkadot-developers/polkadot-docs/pull/2/files